csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs

```diff
@@ -605,6 +688,80 @@ public static OrtValue CreateTensorValueFromMemory<T>(T[] data, long[] shape) wh
             return OrtValue.CreateTensorValueFromMemory(OrtMemoryInfo.DefaultInstance, new Memory<T>(data), shape);
         }
 
+#if NET8_0_OR_GREATER
+#pragma warning disable SYSLIB5001 // System.Numerics.Tensors is only in preview so it can continue receiving API feedback
+        /// <summary>
+        /// This is a factory method creates a native Onnxruntime OrtValue containing a tensor.
+        /// The method will attempt to pin managed memory so no copying occurs when data is passed down
+        /// to native code.
+        /// </summary>
+        /// <param name="value">Tensor object</param>
+        /// <param name="elementType">discovered tensor element type</param>
+        /// <returns>And instance of OrtValue constructed on top of the object</returns>
+        public static OrtValue CreateTensorValueFromDotnetTensorObject<T>(DotnetTensors.ITensor<DotnetTensors.Tensor<T>, T> tensor) where T : unmanaged
+        {
+            if (!IsContiguousAndDense(tensor))
+            {
+                var newTensor = DotnetTensors.Tensor.Create<T>(tensor.Lengths);
+                tensor.CopyTo(newTensor);
+                tensor = newTensor;
+            }
+            unsafe
+            {
+                var backingData = (T[])tensor.GetType().GetField("_values", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(tensor);
```

Comment:
I was having a look at @tjwald's project and saw he has a clever trick for this: https://github.com/tjwald/high-perf-ML/blob/d5054ca0bf882570e59a5d960d45e34b64b81b5d/ML.Infra/TensorExtensions.cs#L28

It seems that generic support like this might only work in .NET 9 though, based on the docs https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafeaccessorattribute?view=net-9.0

Reply:
One question about the compatibility worries me. We support Linux, MacOS, Android and iOS with the same codebase. 
Typically, we looked at NETSTANDARD compatibility for our library.
Now with a transition to NET monikers it is sometimes hard to say what compatible with what.

Reply:
> And i think the other extensions he has there would be something good to discuss

@michaelgsharp I am not sure where to discuss your comment (maybe [here](https://github.com/microsoft/semantic-kernel/issues/9793)?)

Regarding the other extension:

`public static Span<T> GetRowSpan<T>(this TensorSpan<T> tensor, int i)`

It is only necessary because the tensor primitives don't support row by row operations. 
I needed to run a softmax on each row of a tensor, but the softmax in tensor primitives does it on the whole tensor and not row by row.

Reply:
> One question about the compatibility worries me. We support Linux, MacOS, Android and iOS with the same codebase.
> Typically, we looked at NETSTANDARD compatibility for our library.
> Now with a transition to NET monikers it is sometimes hard to say what is compatible with what.

The only thing .NETStandard gives over .NET targeting is support for .NETFramework.  You don't lose that by _adding_ more targetframeworks - you can just specialize your builds that target newer frameworks.